packages/smart-contracts/test/contracts/ERC20EscrowToPay.test.ts

```diff
@@ -0,0 +1,187 @@
+            .withArgs(ethers.utils.keccak256(referenceExample3));
+
+            const requestMapping = await erc20EscrowToPay.connect(payer).requestMapping(referenceExample3);
+            expect(requestMapping.amount).to.be.equal(1000);
+            expect(requestMapping.emergencyState).to.be.true;
+        });
+        it("Should revert if claimed before claimDate.", async () => {
+            await testERC20.connect(payer).approve(erc20EscrowToPayAddress, 1001);
+            await erc20EscrowToPay.connect(payer).payEscrow( testERC20.address, payeeAddress, 1000, referenceExample4, 1, feeAddress);
+            await erc20EscrowToPay.connect(payee).initiateEmergencyClaim(referenceExample4);
+            
+            expect(erc20EscrowToPay.connect(payee).completeEmergencyClaim(referenceExample4)).to.be.reverted;
+            const requestMapping = await erc20EscrowToPay.connect(payer).requestMapping(referenceExample4);
+
+            expect(requestMapping.amount).to.be.equal(1000);
+            expect(requestMapping.emergencyState).to.be.true;
+        });
+    });
+    describe("Cancel emergency flow:", () => {
+        it("Should cancel emergencyClaim, reset state and emit event.", async () => {
+            await testERC20.connect(payer).approve(erc20EscrowToPayAddress, 2002);
+            await erc20EscrowToPay.connect(payer).payEscrow(testERC20.address, payeeAddress, 2000, referenceExample5, 2, feeAddress);
+            await erc20EscrowToPay.connect(payee).initiateEmergencyClaim(referenceExample5)
+            
+            expect(await erc20EscrowToPay.connect(payer).revertEmergencyClaim(referenceExample5))
+            .to.emit(erc20EscrowToPay, "RevertedEmergencyClaim")
+            .withArgs(ethers.utils.keccak256(referenceExample5));
+            
+            const requestMapping = await erc20EscrowToPay.connect(payer).requestMapping(referenceExample5);
+            expect(requestMapping.emergencyState).to.be.false;
+            expect(requestMapping.emergencyClaimDate).to.equals(0);
+        });
+        it("Should revert if emergencyState is canceled.", async () => {
+            await testERC20.connect(payer).approve(erc20EscrowToPayAddress, 2002);
+            await erc20EscrowToPay.connect(payer).payEscrow(testERC20.address, payeeAddress, 2000, referenceExample6, 2, feeAddress);
+            await erc20EscrowToPay.connect(payee).initiateEmergencyClaim(referenceExample6)
+            await erc20EscrowToPay.connect(payer).revertEmergencyClaim(referenceExample6)
+        
+            expect(erc20EscrowToPay.connect(payee).completeEmergencyClaim(referenceExample6)).to.be.reverted;
+
+            const requestMapping = await erc20EscrowToPay.connect(payer).requestMapping(referenceExample6);
+            expect(requestMapping.amount).to.be.equal(2000);
+            expect(requestMapping.emergencyState).to.be.false;
+        });
+    });  
+    describe("Freeze request flow:", () => {
```

Comment:
I think you miss the test for post-freeze period withdrawal by the payer

Reply:
Done. 
I added tests:
line 166:  `it("Should revert if executed before unlockDate."`
line 173: `it("Should revert if try to execute EmergencyClaim while request is frozen."`